### PR TITLE
LESQ-969: Implement OakPagination component on OWA with SEO

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1993 - https://www.youtube.com/watch?v=zkHOVJINRD8

- Create and implement `PaginationHead` Component to add pagination metadata
- Imports OakPagination from CompLib and adds to Unit, Lessons, Search and Blogs/Webinar pages
- Amends usePagination hook, to return props for OakPagination component
- Updates test files props in line with OakPagination  comp
- Removes old Pagination component from OWA

## Issue(s)

Fixes #LESQ-969

## How to test

1. Go to https://deploy-preview-2664--oak-web-application.netlify.thenational.academy

2. Navigate to any page with pagination i.e. Search, Blogs/Webinars, certain lesson and unit pages
https://deploy-preview-2664--oak-web-application.netlify.thenational.academy/webinars?page=1
https://deploy-preview-2664--oak-web-application.netlify.thenational.academy/teachers/programmes/english-secondary-ks3-l/units
https://deploy-preview-2664--oak-web-application.netlify.thenational.academy/teachers/search?term=science

3. You should see new `OakPagination` component has been implemented with smooth scrolling functionality

4. `PaginationHead` component adds next and prev into the head of page when previous or next pages are available 

## SEO Docs

https://www.notion.so/oaknationalacademy/SEO-recommendations-for-pagination-d69ae43e60be4485b3e061d76d0f4542

## Screenshots

How it used to look (delete if n/a):

https://github.com/user-attachments/assets/e8103a95-d1d3-4dfd-92fc-069e71062de8

Screenshot 2024-08-05 at 11 52 42
Screenshot 2024-08-05 at 11 53 20

How it should now look:

Screenshot 2024-08-05 at 11 43 23

https://github.com/user-attachments/assets/d48616af-7cfa-4871-a741-dcd52172f07f

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherComponents/LessonList/LessonList.tsx [hunk 1]
```diff
@@ -48,6 +50,10 @@ const LessonList: FC<LessonListProps> = (props) => {
     onClick,
   } = props;
   const { currentPage, pageSize, firstItemRef } = paginationProps;
+
+  const router = useRouter();
+
+  const paginationRoute = router.asPath.split("?")[0] || router.asPath;
```
Comment (njblore) [inline src/components/TeacherComponents/LessonList/LessonList.tsx:56]: could we return this and the onPageChange fn from pagination props?

File: src/components/TeacherComponents/LessonList/LessonList.tsx [hunk 2]
```diff
@@ -78,8 +84,33 @@ const LessonList: FC<LessonListProps> = (props) => {
...
+        <Box
+          $width="100%"
+          $mt={[0, "auto"]}
+          $pb={[30, 44]}
+          $pt={[46, 36]}
+          data-testid="pagination-box"
+        >
+          <OakPagination
+            {...paginationProps}
+            initialPage={currentPage}
```
Comment (njblore) [inline src/components/TeacherComponents/LessonList/LessonList.tsx:96]: is this the same thing with 2 different names? if so can we rename in one place so it's consistent?

**Review by RBrownsett — APPROVED** [review]
Approved - tested, checked mobile and desktop view along with SEO data. Further testing details on the Notion ticket

File: src/components/TeacherComponents/LessonList/LessonList.tsx (hunk 1 above)
Comment (k-huggs) [reply src/components/TeacherComponents/LessonList/LessonList.tsx]: Moved both into usePagination hook ✅

File: src/components/TeacherComponents/LessonList/LessonList.tsx (hunk 2 above)
Comment (k-huggs) [reply src/components/TeacherComponents/LessonList/LessonList.tsx]: Removed `initialPage` prop so component only takes currentPage, no internal regulation of state ✅

File: src/components/SharedComponents/PostList/PostList.tsx [hunk 3]
```diff
@@ -97,7 +96,12 @@ const PostList: FC<PostListProps> = (props) => {
       ) : null}
       {withPagination && (
         <Box $width="100%" $mt={[0, "auto"]} $pt={48}>
-          <Pagination pageName={blogsOrWebinars} {...paginationProps} />
+          <OakPagination
+            {...paginationProps}
+            onPageChange={onPageChange}
```
Comment (njblore) [inline src/components/SharedComponents/PostList/PostList.tsx:101]: tiny thing but `onPageChange` is coming from the paginationProps now so doesn't need to be added separately

**Review by njblore — APPROVED** [review]
looking good :)

File: src/components/SharedComponents/PostList/PostList.tsx (hunk 3 above)
Comment (k-huggs) [reply src/components/SharedComponents/PostList/PostList.tsx]: Refactored ✅

**Pull request merged**